.github/workflows/test_branches.yml

```diff
@@ -692,71 +692,49 @@ jobs:
         else
             SHA=$GITHUB_SHA
         fi
-        function upload {
-            echo ""
-            echo "Build group: $1"
-            export CODECOV_NAME=$1
-            export TAG=$1
-            export GHA_OS_NAME=${{matrix.TARGET}}
-            shift
-            rm -vf .coverage coverage.xml
-            echo "    $@" | sed 's/ /\n    /'
-            coverage combine --debug=dataio $@
-            if test ! -e .coverage; then
-                echo "No coverage to upload."
-                return
-            fi
-            coverage xml || coverage xml -i
-            i=0
-            while : ; do
-                ((i+=1))
-                rm -f codecov.log
-                echo "Uploading coverage to codecov (attempt ${i})"
-                bash $CODECOV -Z -e TAG,GHA_OS_NAME -X gcov -X s3 \
-                    -f coverage.xml -C $SHA | tee codecov.log
-                if test $? == 0; then
-                    echo "PASS $CODECOV_NAME" >> codecov.result
-                    break
-                elif test $i -ge 2; then
-                    if test `grep successfully codecov.log | wc -l` -gt 0; then
-                        echo "PASS $CODECOV_NAME (implied)" >> codecov.result
-                    else
-                        # Do not fail the build (yet) just because the
-                        # codecov upload fails
-                        echo "FAIL $CODECOV_NAME" >> codecov.result
-                        break
-                    fi
-                fi
-                DELAY=$(( RANDOM % 30 + 30))
-                echo "Pausing $DELAY seconds before re-attempting upload"
-                sleep $DELAY
-            done
-        }
         for ARTIFACT in artifacts/*_*${{matrix.TARGET}}_*; do
             NAME=`echo $ARTIFACT | cut -d/ -f2`
             cp -v $ARTIFACT/.coverage .coverage-$NAME
         done
-        upload ${{matrix.TARGET}} .coverage-*_${{matrix.TARGET}}-*
-        if compgen -G ".coverage-*" > /dev/null; then
-            upload ${{matrix.TARGET}}/other .coverage-*
+        rm -vf .coverage coverage.xml
+        if [ ${{ matrix.TARGET }} == 'linux' ]; then
+          echo "Build: linux/other"
+          echo ""
+          FILES=.coverage-*_other-*
+          coverage combine --debug=dataio $FILES
+          if test ! -e .coverage; then
+              echo "No coverage to upload."
+          else
+              coverage xml || coverage xml -i
+              mv -v coverage.xml coverage-other.xml
+          fi
```

Comment:
Can we do this for all platforms?  Right now we only use Linux for "other", but that may not always be the case.